org.eclipse.scout.rt.dataobject.test/src/test/java/org/eclipse/scout/rt/dataobject/migration/AbstractRemoveDoEntityContributionValueMigrationHandlerTest.java

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) BSI Business Systems Integration AG. All rights reserved.
+ * http://www.bsiag.com/
+ */
+package org.eclipse.scout.rt.dataobject.migration;
+
+import static org.junit.Assert.*;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.scout.rt.dataobject.DoEntity;
+import org.eclipse.scout.rt.dataobject.IDataObjectMapper;
+import org.eclipse.scout.rt.dataobject.IDoEntity;
+import org.eclipse.scout.rt.dataobject.ITypeVersion;
+import org.eclipse.scout.rt.dataobject.fixture.FirstSimpleContributionFixtureDo;
+import org.eclipse.scout.rt.dataobject.fixture.SecondSimpleContributionFixtureDo;
+import org.eclipse.scout.rt.dataobject.fixture.SimpleFixtureDo;
+import org.eclipse.scout.rt.dataobject.migration.DataObjectMigrator.DataObjectMigratorResult;
+import org.eclipse.scout.rt.dataobject.migration.fixture.version.AlfaFixtureTypeVersions.AlfaFixture_1;
+import org.eclipse.scout.rt.dataobject.migration.fixture.version.AlfaFixtureTypeVersions.AlfaFixture_2;
+import org.eclipse.scout.rt.platform.BEANS;
+import org.eclipse.scout.rt.platform.util.CollectionUtility;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Tests {@link AbstractRemoveDoEntityContributionValueMigrationHandler}
+ */
+public class AbstractRemoveDoEntityContributionValueMigrationHandlerTest {
+
+  protected static IDataObjectMapper s_dataObjectMapper;
+
+  @BeforeClass
+  public static void beforeClass() {
+    s_dataObjectMapper = BEANS.get(IDataObjectMapper.class);
+  }
+
+  /**
+   * Tests that data object is not changed if there is no contribution.
+   */
+  @Test
+  public void testMigrationOfDoEntityWithoutContribution() {
+    SimpleFixtureDo fixture = createSimpleFixtureDo();
+
+    String fixtureJson = s_dataObjectMapper.writeValue(fixture);
+    DataObjectMigrationContext ctx = BEANS.get(DataObjectMigrationContext.class).putGlobal(BEANS.get(DoValueMigrationIdsContextData.class).withAppliedValueMigrationIds(getAppliedValueMigrations()));
+    DataObjectMigratorResult<SimpleFixtureDo> result = BEANS.get(DataObjectMigrator.class).migrateDataObject(ctx, fixtureJson, SimpleFixtureDo.class);
```

Comment:
Value migrations act on typed data objects. No need to serialize to string and deserialize again. Use TestDataObjectMigrator instead of DataObjectMigrator